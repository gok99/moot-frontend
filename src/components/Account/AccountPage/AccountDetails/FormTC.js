import React from 'react';
import { Col, Row, Button } from 'react-bootstrap';
 
import '../../../Styles/styles.css';
import '../../account.css';

const FormTC = (props) => {
  const teleUser = props.teleUser;
  const id = props.id;
  
  return (
    <Row>
      <Col>
        <Row>
          <p className="text-account">Telegram Connection</p>
        </Row>
        <Row className="mt-2">
          { teleUser === ''
              ? <>
                  <p className="text-account-sub">You are not connected to telegram yet. Please use the command (/r {id}) with our telegram bot to register. Thank you!</p>
                  <Button className="btn-accountform mt-3" href={"https://t.me/mootapp_bot"} target="_blank" rel="noopener noreferrer">Connect to Telegram</Button>
                </>
              : <p className="text-account-sub">You are currently connected to telegram!</p>
          }
        </Row>
      </Col>
    </Row>
  );
};

export default FormTC;