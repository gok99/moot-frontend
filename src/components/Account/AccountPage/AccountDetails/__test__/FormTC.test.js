import React from 'react';
import ReactDOM from 'react-dom';
import FormTC from '../FormTC';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<FormTC id={0} teleUser={""} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<FormTC id={0} teleUser={""} />).toJSON();
  expect(tree).toMatchSnapshot();
});