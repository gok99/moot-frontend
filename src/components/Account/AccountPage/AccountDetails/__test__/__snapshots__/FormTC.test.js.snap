// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      className="row"
    >
      <p
        className="text-account"
      >
        Telegram Connection
      </p>
    </div>
    <div
      className="mt-2 row"
    >
      <p
        className="text-account-sub"
      >
        You are not connected to telegram yet. Please use the command (/r 
        0
        ) with our telegram bot to register. Thank you!
      </p>
      <a
        className="btn-accountform mt-3 btn btn-primary"
        href="https://t.me/mootapp_bot"
        onClick={[Function]}
        onKeyDown={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Connect to Telegram
      </a>
    </div>
  </div>
</div>
`;
